pkg/components/rook/component.go

```diff
@@ -91,3 +137,13 @@ func (c *component) Metadata() components.Metadata {
 		Namespace: c.Namespace,
 	}
 }
+
+func convertNodeSelector(m map[string]string) string {
```

Comment:
As said, if we have a type alias on `map[string]string`, we can assign this method to it. With this, this is not re-usable.

Reply:
> If `util.NodeSelector` is really being used for `nodeAffinity`, maybe we can rename it and add new `NodeSelector` struct?

What is the purpose of adding something that is not useful underneath? The chart does not expose any `NodeAffinity` then what is the use of exposing it to the user?

> Given that `nodeSelector` is a very common thing for components (or at least I would expect every component to support it eventually), I think it would be really nice to use some structured type for it, even if it's `map[string]string` underneath, as it allows you to implement both validation method and rendering helpers, which will simplify the templating.

What kind of validation would you like to put into place?

---

We use `nodeSelector` as it is intended to be used as key value pair. 

```yaml
nodeSelector:
  {{- range $key, $value := .NodeSelectors }}
  {{ $key }}: "{{ $value }}"
  {{- end }}
{{- end }}
```

But some fields demand for `nodeAffinity`, so the function `convertNodeSelector` that converts the map of `nodeSelector` to the string. The string format here is specific to this helm chart. This is unconventional string format.

```
provisionerNodeAffinity: key1=value1,value2; key2=value3
```

Widely used string conversion of map is of the format:

```
selector: key1=value1,key2=value2
```